src/software/world/field.cpp

```diff
@@ -71,18 +71,39 @@ Rectangle Field::friendlyDefenseArea() const
         Point(-field_x_length_ * 0.5 + defense_x_length_, -defense_y_length_ / 2.0));
 }
 
+Rectangle Field::friendlyDefenseAreaToBoundary() const
+{
+    return Rectangle(
+        Point(-totalXLength() * 0.5, defense_y_length_ / 2.0),
+        Point(-field_x_length_ * 0.5 + defense_x_length_, -defense_y_length_ / 2.0));
+}
+
 Rectangle Field::enemyDefenseArea() const
 {
     return Rectangle(
         Point(field_x_length_ * 0.5, defense_y_length_ / 2.0),
         Point(field_x_length_ * 0.5 - defense_x_length_, -defense_y_length_ / 2.0));
 }
 
+Rectangle Field::enemyDefenseAreaToBoundary() const
+{
+    return Rectangle(
+        Point(totalXLength() * 0.5, defense_y_length_ / 2.0),
+        Point(field_x_length_ * 0.5 - defense_x_length_, -defense_y_length_ / 2.0));
+}
+
 Rectangle Field::friendlyHalf() const
 {
     return Rectangle(friendlyCornerNeg(), Point(0, friendlyCornerPos().y()));
 }
 
+Rectangle Field::friendlyHalfToBoundary() const
+{
+    Point neg_x_neg_y_corner(-totalXLength() / 2, -totalYLength() / 2);
```

Comment:
2 suggestions.

1. We already have a `friendlyCornerPos()` function that can be used here
2. Perhaps even better, why not get the `friendlyHalf()` and expand it by the boundary amount?